pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarSplit.java

```diff
@@ -57,8 +56,7 @@ public PulsarSplit(
             @JsonProperty("schemaName") String schemaName,
             @JsonProperty("tableName") String tableName,
             @JsonProperty("splitSize") long splitSize,
-            @JsonProperty("schema") String schema,
-            @JsonProperty("schemaType") SchemaType schemaType,
+            @JsonProperty("schema") SchemaInfo schemaInfo,
```

Comment:
@sijie is this change necessary and are you sure it json serialization works here for SchemaInfo?  SchemaInfo info is a protobuf and JSON serializers don't work well with protobuf objects

Reply:
SchemaInfo is a POJO not a protobuf generated class.

Reply:
The reason I use `SchemaInfo` is we need all the information for constructing the right schema. A schema info is comprised of type, schema data and properties.  The presto implementation ignores the `properties` field which can potentially create bugs in future. So the change to use `SchemaInfo` is needed.

Reply:
my concern is that one of the fields ```byte[] schema```. JSON don't usually contain bytes and there could be issues with character encoding that is why I passed the schema in as a String in the first place.

Reply:
> The presto implementation ignores the properties field which can potentially create bugs in future.
can you elaborate?

Reply:
If Presto is using Jackson (from the class, it appears to be using Jackson), Jackson handles `byte[]` properly as it use Base64 for encoding the bytes.

If you have concerns about this, I can change it to pass both `SchemaInfo` and `byte[]` and use Base64 for encoding. The current approach doesn't handle string encoding properly.

Reply:
I am fine with passing in SchemaInfo if all the encoding works properly.  For debugging purposes, can you adjust the toString() method so that the binary field ```schema``` in SchemaInfo gets printed out as a String.

Reply:
> can you elaborate?

I mean current implementation only stores type and schema data in the split. It drops properties. This would cause the presto implementation ignores the metadata we attached to properties for a schema, which can create bugs. so whether we flatten all the fields from schema info or use schema info directly doesn't matter here. the point is we need all the fields in schema info.